include/map.h

```diff
@@ -0,0 +1,79 @@
+#ifndef MAP_H
+#define MAP_H
+
+#include "ultra64.h"
+
+struct PlayState;
+
+typedef enum FloorID {
+    /*  1 */ F_8F = 1,
+    /*  2 */ F_7F,
+    /*  3 */ F_6F,
+    /*  4 */ F_5F,
+    /*  5 */ F_4F,
+    /*  6 */ F_3F,
+    /*  7 */ F_2F,
+    /*  8 */ F_1F,
+    /*  9 */ F_B1,
+    /* 10 */ F_B2,
+    /* 11 */ F_B3,
+    /* 12 */ F_B4,
+    /* 13 */ F_B5,
+    /* 14 */ F_B6,
+    /* 15 */ F_B7,
+    /* 16 */ F_B8
+} FloorID;
+
+// All arrays pointed in this struct are indexed by "map indices"
+// In dungeons, the map index corresponds to the dungeon index (which also indexes keys, items, etc)
+// In overworld areas, the map index corresponds to the overworld area index (spot 00, 01, etc)
+typedef struct MapData {
+    /* 0x00 */ s16 (*floorTexIndexOffset)[8]; // dungeon texture index offset by floor
+    /* 0x04 */ s16*  bossFloor; // floor the boss is on
+    /* 0x08 */ s16 (*roomPalette)[32]; // map palette by room
+    /* 0x0C */ s16*  maxPaletteCount; // max number of palettes in a same floor
+    /* 0x10 */ s16 (*paletteRoom)[8][14]; // room by palette by floor
+    /* 0x14 */ s16 (*roomCompassOffsetX)[44]; // dungeon compass icon X offset by room
+    /* 0x18 */ s16 (*roomCompassOffsetY)[44]; // dungeon compass icon Y offset by room
```

Comment:
lately we;ve been removing field name alignment in structs across the board, if you feel like doing it. it looks especially awkward in this struct with some lines starting with ( imo